xlgui/widgets/playlist.py

```diff
@@ -1249,6 +1249,21 @@ def on_drag_data_received(self, widget, context, x, y, selection, info, etime):
         else:
             insert_position = -1
 
+        if self._hack_is_osx:
+            source_widget = Gtk.drag_get_source_widget(context)
+            if (not (source_widget is None or source_widget is self) and
+                    hasattr(source_widget, 'get_selected_tracks')):
+                try:
+                    tracks = source_widget.get_selected_tracks()
+                    if insert_position >= 0:
+                        self.playlist[insert_position:insert_position] = tracks
+                    else:
+                        self.playlist.extend(tracks)
+
+                    return
+                except:
```

Comment:
Maybe do a `logger.debug("error processing DnD hack", exc_info=True)`. Also, `except Exception:` instead.

Reply:
It's only a precaution, like a get_selected_tracks for an object with a different method signature.

We don't actually have any error, and if has, it will be treated as the actual state.

I don't see this as impediment.